src/Microsoft.AspNetCore.Server.Kestrel.Core/KestrelServer.cs

```diff
@@ -272,20 +264,18 @@ private void StartLocalhost(KestrelEngine engine, ServerAddress parsedAddress)
                     Scheme = parsedAddress.Scheme,
                 };
 
-                _disposables.Push(engine.CreateServer(ipv4ListenOptions));
+                var transport = _transportFactory.Create(ipv4ListenOptions, connectionHandler);
+                _transports.Add(transport);
+                transport.BindAsync().Wait();
```

Comment:
For backwards compatibility we do. BindAsync() can throw AddressInUse exceptions for example.